packages/apps/plugins/plugin-grid/src/components/GridMain.tsx

```diff
@@ -6,9 +6,9 @@ import React, { forwardRef, type FC } from 'react';
 
 import { DocumentType, GridItemType, type GridType } from '@braneframe/types';
 import { Surface, parseMetadataResolverPlugin, useResolvePlugin } from '@dxos/app-framework';
-import { type EchoReactiveObject, isEchoObject } from '@dxos/echo-schema';
+import { type EchoReactiveObject } from '@dxos/echo-schema';
```

Comment:
I think our goal with the split was to separate everything type-definition-related from data management dependencies. 

EchoReactiveObject is a type, implementation lives in database package, so a function that is able to tell whether an object is an echo object lives there, cause it relies on the implementation.

even non-echo reactive objects need to know about ids in schema because id is generated when we create an object, not when we add it to a database. it's another topic whether it's correct, but quite a number of functionality deals with ids.

so echo-schema isn't supposed to know nothing about database concepts (hence the echo- prefix), it's just supposed to be decoupled from data-manipulations, storage details etc

Reply:
Yeah fair, since they're types it definitely seems not as bad/leaky. I'm still a little skeptical, but either way I think getting all the implementations out of echo-schema is win so I'm happy with this 👍